### PR TITLE
arrayop.d: color syntax highlighting

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/ddmd/expression.d [hunk 1]
```diff
@@ -11007,10 +11007,7 @@ extern (C++) final class NegExp : UnaExp
         if (tb.ty == Tarray || tb.ty == Tsarray)
         {
             if (!isArrayOpValid(e1))
-            {
-                error("invalid array operation %s (possible missing [])", toChars());
-                return new ErrorExp();
-            }
+                return arrayOpInvalidError(this);
```
Comment (braddr) [inline src/ddmd/expression.d]: A thought.. this feels like the wrong refactoring.  Consider moving the error() call to inside isArrayOpValid and changing this return to just "return new ErrorExp();"

Comment (WalterBright) [inline src/ddmd/expression.d]: This maintains the simple true/false result of `isArrayOpValid()` and the purity of it making it simpler to understand.

Comment (ZombineDev) [inline src/ddmd/expression.d]: I agree with @WalterBright.

**Review by PetarKirov — APPROVED** [review]
Looks good to me.

File: src/ddmd/expression.d (hunk 1 above)
Comment (CyberShadow) [inline src/ddmd/expression.d]: The refactoring seems fine, but it should have been done in a separate commit or PR.

**Comment by JinShil** [comment]
Not sure about the code changes, but the error messages look good to me.

**Comment by andralex** [comment]
This is an entirely appropriate PR. Generally, we're experiencing a deflation of reviewers relative to PRs so I recommend favoring moving things along over minor blocking suggestions. Thanks.

**Review by andralex — APPROVED** [review]

**Comment by CyberShadow** [comment]
@andralex No worries, @yebblies merges these regularly when he gets the chance to look at them.

**Comment by WalterBright** [comment]
@cybershadow It's 18 days old. There are approx 1200 error() calls in dmd. Waiting 18 days to do a handful means they'll never get syntax highlighted.

**Comment by CyberShadow** [comment]
> WalterBright requested a review from yebblies 20 hours ago

This isn't 18 days old though ;)

**Comment by wilzbach** [comment]
> continuous-integration/jenkins/pr-merge — This commit cannot be built

CC @CyberShadow @MartinNowak - third request for escalation against the random build failures at the project-tester ;-)

**Comment by CyberShadow** [comment]
I'll disable Jenkins tomorrow morning if nothing new happens.

**Comment by yebblies** [comment]
>> WalterBright requested a review from yebblies 20 hours ago
> This isn't 18 days old though ;)

Exactly.  And @andralex knows how to contact me if there's something urgent.
